### PR TITLE
Add kustomize as deployment template engine

### PR DESCRIPTION
**Pull request by xiabai84**
Signed-off-by: xiabai84 <xiabai84@gmail.com>

### CODE REVIEW

**Comment by xiabai84** [comment]
fix #41

**Comment by xiabai84** [comment]
/assign @linuxsuren

My contribution seems like OK, but there are some failed sub-tasks...

**Comment by JohnNiang** [comment]
> /assign @LinuxSuRen
> 
> My contribution seems like OK, but there are some failed sub-tasks...

Don't worry about the unstable CI checks. We will inspect that later.

File: kustomize/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM kubespheredev/builder-base:v3.1.0
+
+ENV KUSTOMIZE_VER 4.3.0
+
+RUN cd && \
```
Comment (JohnNiang) [inline kustomize/Dockerfile:5]: Please refer to this hack file: https://github.com/kubesphere/devops-agent/blob/master/base/hack/install_utils.sh.

IMO, there is no need a separated docker image for users to use `kustomize`.

**Review by JohnNiang — CHANGES_REQUESTED** [review]

File: kustomize/Dockerfile (hunk 1 above)
Comment (xiabai84) [reply kustomize/Dockerfile:5]: Do you mean, kustomize should be included within the base-image?

Comment (xiabai84) [reply kustomize/Dockerfile:5]: Indeed, kustomize is already integrated into kubectl. Please close my issue.

**Comment by JohnNiang** [comment]
Thanks for your contribution! I'm going to close this PR due to https://github.com/kubesphere/devops-agent/pull/42#discussion_r715952148.
/close

**Comment by JohnNiang** [comment]
I'm going to reopen this PR due to https://github.com/kubesphere/devops-agent/issues/41#issuecomment-1069156295
/reopen

File: kustomize/Dockerfile (hunk 1 above)
Comment (LinuxSuRen) [reply kustomize/Dockerfile:5]: hi @xiabai84 ， I'm wondering if we can put `kustomize` into `builder-base`

**Comment by LinuxSuRen** [comment]
hi @xiabai84, we intend to use `kustomize` command in the next release. I might do it in next week if you don't have free time to do it.

**Comment by xiabai84** [comment]
> hi @xiabai84, we intend to use `kustomize` command in the next release. I might do it in next week if you don't have free time to do it.

Hi @LinuxSuRen, I'm so glad to hear, that kustomize will be provided in the next release. This #42 is actually about build kustomize as new extension. Maybe we should open a new PR for extending "base" container with kustomize. I think I can do that. So should I open a new PR?

**Comment by JohnNiang** [comment]
@xiabai84 Please feel free to submit a new PR.

**Pull request closed without merging**